### ISSUE TITLE
[rllib] Huge performance difference between training and evaluation

### ISSUE THREAD

**Issue by nlztrk** [ray-project/ray#15079]
### What is the problem?
During the training phase the agent seems to have completely learned to solve the environment. But when I restore it and run a simple evaluation run, I see very low episode rewards.

*Ray version and other system information (Python version, TensorFlow version, OS):*
Tried Ray 0.9.0, 1.2.0 and 2.0.0
Python 3.7
Pytorch 1.7.1
Tensorflow 2.4.1

### Reproduction (REQUIRED)

Simple training loop for 20 iterations
```
config = ppo.DEFAULT_CONFIG.copy()
config["framework"] = "torch"
config["evaluation_config"] = {"explore":False}

iterations = 20
agent = ppo.PPOTrainer(config=config, env="CartPole-v0")
checkpoint = None
train_results = {}

for i in range(iterations):
    train_results = agent.train()
    print(i, train_results['episode_reward_mean'], train_results['episodes_total'])
    checkpoint = agent.save("sanity")
agent.stop()
```
Getting this output for training:
```
0 22.421348314606742 178
1 42.54 264
2 68.19 301
3 97.42 326
4 126.01 347
5 155.73 367
6 174.89 389
7 188.39 409
8 194.93 429
9 196.44 449
10 196.8 469
11 200.0 489
12 200.0 509
13 200.0 529
14 200.0 549
15 200.0 569
16 200.0 589
17 200.0 609
18 200.0 629
19 200.0 649
```

and evaluating with:
```
eval_agent = ppo.PPOTrainer(config=config, env="CartPole-v0")
eval_agent.restore(checkpoint)

env = gym.make("CartPole-v0")

ep_rews = []

for i in range(100):

    obs = env.reset()
    done = False
    ep_rew = 0

    while not done:  
        action = eval_agent.compute_action(obs)
        next_obs, reward, done, info = env.step(action)
        ep_rew += reward
    ep_rews.append(ep_rew)

env.close()
np.mean(ep_rews)
```
And getting mean reward of **~15-20**

It remains whether I select Torch or TF and CPU-Only or GPU.

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by nlztrk** [comment]
That was a mistake I made while implementing it quickly. I forgot to set the name of the observation variable correctly.
